### PR TITLE
Integrated report to the backoffice

### PR DESCRIPTION
**Pull request by Omri-Levy**
## **User description**
- feat(*): integrated the report pdf into the backoffice
- refactor(*): made reports endpoint return one report

### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

___

## **Type**
enhancement, bug_fix

___

## **Description**
- Added functionality to fetch and display business reports in the backoffice application.
- Enhanced website monitoring block to support detailed analysis and reporting.
- Refactored hooks to use the updated query hook for fetching current case data.
- Improved UI responsiveness for case actions.
- Introduced internal controller, service methods, and DTO for handling business reports in the workflow service.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>10 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>fetchers.ts</strong><dd><code>Add Business Reports Fetching Functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/domains/business-reports/fetchers.ts
<li>Added fetchBusinessReports function to fetch business reports with Zod <br>validation.<br> <li> Introduced BusinessReportSchema and BusinessReportsSchema for data <br>validation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-83ca1583ebb91b1a8374249f4220fb0d73925241f95a48ee14f92fb2ee144672">+25/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useLatestBusinessReportQuery.tsx</strong><dd><code>Implement Latest Business Report Query Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/domains/business-reports/hooks/queries/useLatestBusinessReportQuery/useLatestBusinessReportQuery.tsx
<li>Implemented useLatestBusinessReportQuery hook for fetching the latest <br>business report.<br> <li> Utilizes react-query for data fetching and caching.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-5bd4f4c878a819edb01c008d72b3ae831a2273fd84ce3d27c7c735ea30efebcc">+25/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>query-keys.ts</strong><dd><code>Setup Query Keys for Business Reports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/domains/business-reports/query-keys.ts
<li>Created businessReportsQueryKey using createQueryKeys for react-query.<br> <li> Setup query key for fetching the latest business report.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-1fb0e778a17228933f13bf45da6f637f2d412fb013d09cd65a97a9b1950c3619">+16/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useWebsiteMonitoringBlock.tsx</strong><dd><code>Enhance Website Monitoring Block Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/hooks/useWebsiteMonitoringBlock/useWebsiteMonitoringBlock.tsx
<li>Extended useWebsiteMonitoringBlock to support detailed website <br>monitoring data.<br> <li> Added types for the function parameters and return values.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-5c1e0e85641f1da7760a979303aefd26a4e0895cb6b623a40ec4f0890857ccbf">+48/-1</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>useWebsiteMonitoringReportBlock.tsx</strong><dd><code>Implement Website Monitoring Report Block Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/variants/WebsiteMonitoringBlocks/hooks/useWebsiteMonitoringReportBlock/useWebsiteMonitoringReportBlock.tsx
<li>Introduced useWebsiteMonitoringReportBlock hook for fetching and <br>displaying website monitoring reports.<br> <li> Utilizes useStorageFileByIdQuery for fetching report file URL.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-1c55b1fcce0ff19e7224ee28c5a34ce3c893cfd2c2ca92fb13a7e6ddcc0ee12c">+46/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Case.Actions.tsx</strong><dd><code>Improve Case Actions UI Responsiveness</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/Entity/components/Case/Case.Actions.tsx
<li>Adjusted className for case action titles to support text wrapping.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-f3e71bf810a855cd82f26bfb67e942d73aa95a6001bfaa7c5a5264a2d9daa3bd">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>business-report.controller.internal.ts</strong><dd><code>Add Internal Business Report Controller</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/business-report/business-report.controller.internal.ts
<li>Added BusinessReportControllerInternal with getLatestBusinessReport <br>endpoint.<br> <li> Endpoint fetches the latest business report based on businessId and <br>reportType.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-1706a1a1db1bdcc4460b4d3eebc49da805a6f1c0a7df4fce7424bdb73b8b6456">+36/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>business-report.repository.ts</strong><dd><code>Extend Business Report Repository with findFirst Method</code>&nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/business-report/business-report.repository.ts
<li>Added findFirst method to BusinessReportRepository for querying single <br>reports.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-f7c05788fae503eca95f5d77ec7106ee4b30d0da871d53ce24f883bd4e159972">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>business-report.service.ts</strong><dd><code>Add findFirst Method to Business Report Service</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/business-report/business-report.service.ts
<li>Introduced findFirst method in BusinessReportService to utilize <br>repository's findFirst.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-5c71c9de41e207ba2f090ca025f87d65fc01a53535e73aad6db157a62cd0bbeb">+10/-3</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>get-business-reports.dto.ts</strong><dd><code>Define DTO for Getting Business Reports</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/business-report/get-business-reports.dto.ts
- Created DTO for getting business reports with validation rules.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-b7f31335ed03713e012b9fb71237f1ac86daea81e41f5cf83cb3a356e7393b71">+18/-0</a>&nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Bug_fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>useKYCBusinessInformationBlock.ts</strong><dd><code>Refactor KYC Business Information Block Hook</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/lib/blocks/hooks/useKYCBusinessInformationBlock/useKYCBusinessInformationBlock.ts
- Refactored to use useCurrentCaseQuery instead of useCurrentCase.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-488528fa6d1c57360231accb0c46aad7228ede314dc9f46e5fa40aa021ad0ac8">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>business-report.module.ts</strong><dd><code>Register Internal Business Report Controller</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/business-report/business-report.module.ts
<li>Registered BusinessReportControllerInternal in the <br>BusinessReportModule.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2297/files#diff-dac6e8930b5c228b99d5c53c829798be5d299bc59676ed0cfe69b1ca3262a683">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

## Summary by CodeRabbit

- **New Features**
	- Introduced functionality to fetch business reports by business ID and report type.
	- Added a custom hook for displaying website monitoring PDF reports in a business context.
- **Refactor**
	- Simplified the conditional rendering logic in the `PDFViewerCell` component to improve code readability.
- **Style**
	- Adjusted the styling of the `DefaultBlocks` component to ensure it fills the height of its parent container, enhancing layout consistency across devices.

### CODE REVIEW

**Review by chesterkmr — APPROVED** [review]

File: apps/backoffice-v2/src/pages/Entity/components/Case/Case.Actions.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ export const Actions: FunctionComponent<IActionsProps> = ({
       <div className={`flex h-20 justify-between`}>
         <div className={`flex flex-col space-y-3`}>
           <h2
-            className={ctw(`text-4xl font-semibold leading-9`, {
+            className={ctw(`w-[35ch] break-all text-4xl font-semibold leading-9`, {
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/Case/Case.Actions.tsx:55]: Case actions no longer break visually on long names.

File: apps/backoffice-v2/src/lib/blocks/hooks/useKYCBusinessInformationBlock/useKYCBusinessInformationBlock.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import { useCaseInfoBlock } from '@/lib/blocks/hooks/useCaseInfoBlock/useCaseInfoBlock';
-import { useCurrentCase } from '@/pages/Entity/hooks/useCurrentCase/useCurrentCase';
+import { useCurrentCaseQuery } from '@/pages/Entity/hooks/useCurrentCaseQuery/useCurrentCaseQuery';
 
 export const useKYCBusinessInformationBlock = () => {
-  const { data: workflow } = useCurrentCase();
+  const { data: workflow } = useCurrentCaseQuery();
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/hooks/useKYCBusinessInformationBlock/useKYCBusinessInformationBlock.ts:5]: Boyscout. Renamed hook.

File: apps/backoffice-v2/src/lib/blocks/hooks/useWebsiteMonitoringBlock/useWebsiteMonitoringBlock.tsx
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/hooks/useWebsiteMonitoringBlock/useWebsiteMonitoringBlock.tsx:1]: Boyscout. Added missing types.

**Review by tomer-shvadron — APPROVED** [review]

**Comment by Omri-Levy** [comment]
Should be revisited.

File: apps/backoffice-v2/src/lib/blocks/components/PDFViewerCell/PDFViewer.tsx [hunk 3]
```diff
@@ -4,7 +4,9 @@ import { FunctionComponent } from 'react';
 export const PDFViewerCell: FunctionComponent<TPDFViewerCell> = ({ props, value }) => {
   const { width, height } = props;
 
-  return value ? (
-    <iframe src={`data:application/pdf;base64,${value}#navpanes=0`} width={width} height={height} />
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/components/PDFViewerCell/PDFViewer.tsx:8]: See if #navpanes=0 can work with signed url

Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/components/PDFViewerCell/PDFViewer.tsx:8]: Make sure base64 flow works again.

File: apps/backoffice-v2/src/lib/blocks/variants/DefaultBlocks/DefaultBlocks.tsx [hunk 4]
```diff
@@ -9,7 +9,7 @@ export const DefaultBlocks = () => {
   const { blocks, tabs, activeTab, setActiveTab } = useDefaultBlocksLogic();
 
   return (
-    <div className="relative flex flex-col">
+    <div className="relative flex h-full flex-col">
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/variants/DefaultBlocks/DefaultBlocks.tsx:12]: Make sure changing height does not affect the other tabs.

File: apps/backoffice-v2/src/lib/blocks/variants/DefaultBlocks/DefaultBlocks.tsx [hunk 5]
```diff
@@ -25,7 +25,7 @@ export const DefaultBlocks = () => {
           </div>
         </div>
       )}
-      <div className="flex flex-col gap-4">
+      <div className="flex h-full flex-col gap-4">
         <BlocksComponent blocks={blocks} cells={cells}>
           {(Cell, cell) => <Cell {...cell} />}
         </BlocksComponent>
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/variants/DefaultBlocks/DefaultBlocks.tsx:31]: This is supposed to be inside the `TabsContent` component.

File: apps/backoffice-v2/src/lib/blocks/variants/WebsiteMonitoringBlocks/hooks/useWebsiteMonitoringReportBlock/useWebsiteMonitoringReportBlock.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import { useStorageFileByIdQuery } from '@/domains/storage/hooks/queries/useStorageFileByIdQuery/useStorageFileByIdQuery';
+import { useLatestBusinessReportQuery } from '@/domains/business-reports/hooks/queries/useLatestBusinessReportQuery/useLatestBusinessReportQuery';
+
+export const useWebsiteMonitoringReportBlock = () => {
+  const { data: workflow } = useCurrentCaseQuery();
+  const { data: businessReport } = useLatestBusinessReportQuery({
+    businessId: workflow?.context?.entity?.ballerineEntityId,
+    reportType: 'MERCHANT_REPORT_T1',
+  });
+  const { data: reportFile } = useStorageFileByIdQuery(businessReport?.report?.reportFileId ?? '', {
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/lib/blocks/variants/WebsiteMonitoringBlocks/hooks/useWebsiteMonitoringReportBlock/useWebsiteMonitoringReportBlock.tsx]: Rename `reportFile`.

File: apps/backoffice-v2/src/lib/blocks/variants/DefaultBlocks/DefaultBlocks.tsx (hunk 5 above)
Comment (Omri-Levy) [reply apps/backoffice-v2/src/lib/blocks/variants/DefaultBlocks/DefaultBlocks.tsx:31]: Will be addressed at a later time.

**Pull request merged**